TempFileGZIPWriter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace samdark\sitemap;
+
+/**
+ * Flushes buffer into temporary stream and compresses stream into a file on finish
+ */
+class TempFileGZIPWriter
+    implements WriterInterface
+{
+    /**
+     * @var string Name of target file
+     */
+    private $filename;
+
+    /**
+     * @var ressource for php://temp stream
+     */
+    private $tempFile;
+
+    public function __construct($filename)
+    {
+        $this->filename = $filename;
+        $this->tempFile = fopen('php://temp/', 'w');
```

Comment:
Are there any benefits trying to use `php://memory` stream instead of file?

Reply:
Worth adding `b` to fopen flags in order for it to be binary safe.

Reply:
I don’t think it is a good idea, to always keep everything in memory. It might be possible though to allow changing the point at which a temp file is used. To quote from the manual:

> The memory limit of php://temp can be controlled by appending /maxmemory:NN, where NN is the maximum amount of data to keep in memory before using a temporary file, in bytes. 

Anyway I think is not something to be done in this PR.